app/controllers/UserTokenController.scala

```diff
@@ -188,4 +192,18 @@ class UserTokenController @Inject()(dataSetDAO: DataSetDAO,
       else UserAccessAnswer(granted = false, Some(s"No ${mode.toString} access to tracing"))
     }
   }
+
+  private def handleJobExportAccess(jobId: String, mode: AccessMode, userBox: Box[User]): Fox[UserAccessAnswer] =
+    if (mode != AccessMode.read)
+      Fox.successful(UserAccessAnswer(granted = false, Some(s"Unsupported acces mode for job exports: $mode")))
+    else {
+      for {
+        jobIdValidated <- ObjectId.parse(jobId)
+        jobBox <- jobDAO.findOne(jobIdValidated)(DBAccessContext(userBox)).futureBox
+        answer = jobBox match {
+          case Full(_) => UserAccessAnswer(granted = true)
+          case _       => UserAccessAnswer(granted = false, Some(s"No ${mode} access to job export"))
+        }
+      } yield answer
+    }
```

Comment:
Before this change, the route was directly in wK, so fetching the job from the db checked the access, returning Fox.failure if the user had no access https://github.com/scalableminds/webknossos/blob/master/app/controllers/JobsController.scala#L186 – or at least it would have, had `findOne` been overwritten back then, as I did not. This was a bug.
As pointed out above, the old route and its implementation still need to be removed, which is probably why you did not see it in the diff

Reply:
Perfect, thanks for the explanation :+1: